### PR TITLE
Improvements to automatic cmd help + docs for generate solutions and run evals

### PR DESCRIPTION
**Pull request by Kipok**
Some improvements to automatic dataclass -> cmd help:
1. Added arbitrary kwargs that can be passed into the get_help_message function and can be used to format variables in the help. E.g. it's currently used to provide choices for the datasets as well as extra server arguments that are parsed from BaseModel docs.
2. Added additional help_message parameter that can be used to further customized help with extra instructions on top of auto dataclass docs.
3. Passed in help from generate_solutoins.py into run_evals.py as an example of how we can handle pipeline script docs
4. Added some coloring + a bit improved format for the cmd docs to make arguments more visible.

Please try to see the help for those scripts and let me know if it looks good or if you'd like to change something!
```
python pipeline/run_eval.py --help
python nemo_skills/inference/generate_solutions.py --help
```

I will work on writing docs for other files in the meantime.

### CODE REVIEW

**Comment by Kipok** [comment]
@titu1994 @SeanNaren @shtoshni @VahidooX @i-vainn - FYI in case you have any suggestions or concerns with the way cmd help currently looks like. Please also help to approve if you don't have any concerns.

**Review by titu1994 — COMMENTED** [review]
It looks nice. I wonder if you want to add support for -h too. Not really needed though.

**Comment by i-vainn** [comment]
Thanks @Kipok, now docstring is way more readable!

It works mostly fine, but fails when I run `python nemo_skills/finetuning/prepare_sft_data.py --help` with
```
Traceback (most recent call last):
  File "nemo_skills/finetuning/prepare_sft_data.py", line 210, in <module>
    help_msg = get_help_message(PrepareSFTDataConfig)
  File "nemo_skills/utils.py", line 161, in get_help_message
    docstring = fr'{docstring}'.format(**kwargs)
IndexError: Replacement index 0 out of range for positional args tuple
```
I suppose this happens because `metadata: Dict[Any, Any] = field(default_factory=dict)` line is represented as `metadata: dict[Any, Any] = {}` after parsing and format tries to fill the `{}`.

File: nemo_skills/inference/server/model.py [hunk 1]
```diff
@@ -45,6 +46,27 @@ def remove_stop_tokens(text: str, stop_phrases: List[str]) -> str:
...
+    """Base model class for handling requests to the inference server.
+
+    Args:
+        host: Optional[str] = '127.0.0.1' - Host of the inference server.
+        port: Optional[str] = '5000' - Port of the inference server.
+        sandbox: Optional[Sandbox] = None - Sandbox for executing code.
+            Only required if handle_code_execution is True.
+        ssh_server: Optional[str] = None - SSH server for tunneling requests.
+            Useful if server is running on slurm cluster to which there is an ssh access
+            Can also be specified through SSH_SERVER env var..
```
Comment (i-vainn) [inline nemo_skills/inference/server/model.py:58]: Typo with '..' here and on the line above

**Review by i-vainn — APPROVED** [review]
Looks good, let's merge!

**Pull request merged**